forward_engineering/alterScript/alterScriptFromDeltaHelper.js

```diff
@@ -78,10 +78,41 @@ const getAlterContainersScriptDtos = ({ schema, isUnityCatalogSupports, provider
     ];
 };
 
+/**
+ * @typedef FilterOutExistingStatementsParams
+ * @type {object}
+ * @property {Array<AlterScriptDto>} alterScriptDtos
+ * @property {Set<string>} existingAlterStatements
+ */
+/**
+ * @param {FilterOutExistingStatementsParams} param
+ * @returns {FilterOutExistingStatementsParams}
+ */
+const filterOutExistingStatements = ({ alterScriptDtos, existingAlterStatements }) => {
+	const filteredAlterScriptDtos = alterScriptDtos
+		.filter(Boolean)
+		.flatMap(alterScriptDto => {
+			return alterScriptDto.scripts?.map(scriptDto => {
+				if (existingAlterStatements.has(scriptDto?.script)) {
+					return undefined;
+				}
+				return AlterScriptDto.getInstance([scriptDto.script], alterScriptDto.isActivated, scriptDto.isDropScript);
+			});
+		})
+		.filter(Boolean);
+	const filteredExistingAlterScriptStatements = new Set([
+		...Array.from(existingAlterStatements),
+		...alterScriptDtos.flatMap(dto => dto?.scripts?.map(scriptDto => scriptDto?.script)).filter(Boolean),
+	]);
```

Comment:
```suggestion
	const filteredAlterScriptDtos = alterScriptDtos
		.filter(Boolean)
		.flatMap(alterScriptDto => alterScriptDto.scripts || [])
		.filter(scriptDto => !existingAlterStatements.has(scriptDto?.script))
		.map(scriptDto => AlterScriptDto.getInstance([scriptDto.script], alterScriptDto.isActivated, scriptDto.isDropScript))
		.filter(Boolean);
		
	const allScriptStatements = alterScriptDtos
		.flatMap(dto => dto?.scripts?.map(scriptDto => scriptDto?.script))
		.filter(Boolean);
		
	const filteredExistingAlterScriptStatements = new Set([
		...Array.from(existingAlterStatements),
		...allScriptStatements,
	]);
```

Can we reduce nesting and a bit of complexity?